sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs

```diff
@@ -33,5 +102,349 @@ public void List_Corporate_Actions_For_One_Day()
+            var transactions = new List<TransactionRequest>();
+
+            var currencyLuid = "CCY_GBP";
+
+            // Create the portfolio
+            var request = new CreateTransactionPortfolioRequest(
+                code: portfolioCode,
+                displayName: $"Portfolio-{uuid}",
+                baseCurrency: "GBP",
+                created: startDate,
+                corporateActionSourceId: new ResourceId(
+                    scope: TestDataUtilities.TutorialScope,
+                    code: _corpActionTestSource
+                    )
+            );
+            _transactionPortfoliosApi.CreatePortfolio(TestDataUtilities.TutorialScope, request);
+
+            // Add starting cash position
+            transactions.Add(TestDataUtilities.BuildCashFundsInTransactionRequest(2960000, "GBP", startDate));
+
+            // Add an equity position
+            transactions.Add(TestDataUtilities.BuildTransactionRequest(_instrumentIds[0], 132000, 5, "GBP", startDate, "StockIn"));
+
+            _apiFactory.Api<ITransactionPortfoliosApi>().UpsertTransactions(TestDataUtilities.TutorialScope, portfolioCode, transactions);
+
+            // Upsert Corporate Action   
+            var identifierMapping = new Dictionary<string, string>();
+            identifierMapping.Add("Instrument/default/LusidInstrumentId", _instrumentIds[0]);
+
+            var stockSplitCorpActionRequest = new UpsertCorporateActionRequest(
+                TestDataUtilities.TutorialScope,
+                "Stock Split",
+                announcementDate,
+                exDate,
+                recordDate,
+                paymentDate,
+                new List<CorporateActionTransitionRequest>()
+                {
+                    new CorporateActionTransitionRequest()
+                    {
+                        InputTransition = new CorporateActionTransitionComponentRequest (identifierMapping,1,1),
+                        OutputTransitions = new List<CorporateActionTransitionComponentRequest>()
+                        {
+                            new CorporateActionTransitionComponentRequest(identifierMapping,2,1)
+                        }
+                    }
+                });
+
+            _corporateActionSourcesApi.BatchUpsertCorporateActions(TestDataUtilities.TutorialScope, _corpActionTestSource, new List<UpsertCorporateActionRequest>() { stockSplitCorpActionRequest });
+
+            //
+            // fetch holdings pre ex-dividend date
+            //
+            var holdingsResultPreExDate = _transactionPortfoliosApi.GetHoldings(TestDataUtilities.TutorialScope, portfolioCode, effectiveAt: announcementDate);
+            var holdingsPreExDate = holdingsResultPreExDate.Values.OrderBy(h => h.InstrumentUid).ToList();
+
+            //  check cash balance hasn't changed
+            Assert.That(holdingsPreExDate[0].InstrumentUid, Is.EqualTo(currencyLuid));
+            Assert.That(holdingsPreExDate[0].Units, Is.EqualTo(2960000));
```

Comment:
I wonder whether we need the cash balance check for a stock split?

Reply:
Agreed - removed.